source/common/config/grpc_mux_impl.h

```diff
@@ -89,6 +94,17 @@ class GrpcMuxImpl
   std::unordered_map<std::string, ApiState> api_state_;
   // Envoy's dependency ordering.
   std::list<std::string> subscriptions_;
+  const std::string& server_type_url_;
+  Server::ConfigTracker::EntryOwnerPtr config_tracker_entry_;
+  TimeSource& time_source_;
+  envoy::api::v2::core::GrpcService grpc_service_;
+  // Per Service Control Plane configuration that Envoy is connected to. For services like
+  // RouteDiscoveryService and EndpointDiscovery the configuration allows to specify multiple config
+  // sources, for example a different RouteDiscoveryService can be specified for each listener.
+  static std::unordered_map<
```

Comment:
Static non-PODs are disallowed due to the static initialization fiasco. I'm wondering if static is really the right thing here, since it also might be harder from a testing perspective to reason about. Is there an alternative that isn't too hard to implement instead?

Reply:
I agree it is harder to reason about and test. But could not find any better alternative. Let me know if you have any ideas - I can implement it.

Reply:
Hmm. Can we make `ConfigTracker` smarter somehow? This is essentially responsible for keeping track of all the configs and aggregation, so it seems a natural place to consider doing this. And/or change the shape of the proto data model to make this easier.

Reply:
Thinking more about this, I can think of following options
1. Make `ConfigTracker` smart by introducing some thing similar to `managed_config` which is actually managed by `ConfigTracker` instance - so if this config is declared as `managed_config`, whenever this needs to be changed we can get, update and set it back. When '/config_dump' is called, the `ConfigTracker` can return the config by itself for managed configs instead of calling config dump function call back.
2. Can we just store this specific config in `Singleton Manager` avoiding static here?

Option 2 may be simpler to implement, Option 1 might be a useful addition to `ConfigTracker` framework and can be used in future.

WDYT?

Reply:
What do you mean by `managed_config` in (1)? I think in general, putting the smarts in `ConfigTracker` is the best way to go, so +1.

Reply:
By `managed_config` - I mean the config dump proto structure is owned by `ConfigTracker` and `grpc_mux_impl` would get and update it whenever it needs and when `/config_dump` is called for this type `managed_config` the `ConfigTracker` would just dump what it has instead of calling the callback.

Reply:
Yep, that sounds like a good solution.